### ISSUE TITLE
Please provide unlock password

### ISSUE THREAD

**Issue by turboboost55** [ApopheniaPays/excel-liquidity-pool-simulator#2]
I was hoping to use your spreadsheet as a learning tool for uniswap, but with no documentation and no password to see the formulas, it is not helping me very much.  I realize there is a Show Formulas command in Excel, but in many cases the formula is too wide to fit in the column, and since the sheet is locked I can't widen the column.  Also, with a locked sheet you do not have that very useful feature of being able to view the formula in that 'colored schematic' way.  

Honestly, I don't understand why you locked it.  It would make sense if this was a shared Google doc spreadsheet, but not with an Excel spreadsheet that you have to download.

**Comment by ApopheniaPays** [comment]
Hi, sorry, you sound possibly frustrated and I understand, I've been there. 

I'd have to dig back through my notes to see what my reasoning was at the time for not posting this fully unlocked but IIRC there were a few issues...generally, I really don't like being proprietary about things, I'd rather share and ashare-alike. But it had taken a hell of a lot of work to create this, and right around that time I had been pretty badly taken advantage of by people in the DeFi community, and while I didn't mind sharing a simulation to play with for free I just wasn't feeling so charitable anymore about giving away 100% of my work to a community that seemed at the time to be full of people doing more taking than giving. That's not your fault, obviously, but it's where I was at, and I had originally written this just for my own use, and decided to post it for people to play with if they liked, but that's it. It still works to show how impermanent losses work even though you can't see the formulas (which was the main intent of posting it, if I recall).

I probably should have just posted it on my website, not on Github, since it's not fully open-source. But all my other stuff is here, and it's still a free giveaway in terms of being able to use it if you're familiar enough with Uniswap to understand it. But, from what you're saying, I'd say there are much better ways to learn about Uniswap than trying to untangle the thicket of calculations in this spreadsheet.

A few points, though: there's nothing locked up inside this spreadsheet's code that isn't available in much clearer form already, I just put pieces together that I copied wholesale from public sources. The sole thing you might find in here that might be tough to find is the calculation for how many liquidity pool tokens get issued when you stake something. However, i copied those calculaltions word-for-word from the Uniswap core contracts, and you can find that exact code at [https://github.com/Uniswap/uniswap-v2-core/blob/master/contracts/UniswapV2Pair.sol](https://github.com/Uniswap/uniswap-v2-core/blob/master/contracts/UniswapV2Pair.sol), on lines 119-124:

```
        if (_totalSupply == 0) {
            liquidity = Math.sqrt(amount0.mul(amount1)).sub(MINIMUM_LIQUIDITY);
           _mint(address(0), MINIMUM_LIQUIDITY); // permanently lock the first MINIMUM_LIQUIDITY tokens
        } else {
            liquidity = Math.min(amount0.mul(_totalSupply) / _reserve0, amount1.mul(_totalSupply) / _reserve1);
        }
        require(liquidity > 0, 'UniswapV2: INSUFFICIENT_LIQUIDITY_MINTED');
        _mint(to, liquidity);

```

This says: if you're the first person staking in the pool (the pool creator), the number of liquidity pool tokens is the square root of the product of the quantities of token 1 and token 2, minus a constant minimum liquidity specified elsewhere in the contract. Otherwise you're staking in an existing pool, the number of liquidity  pool tokens you get is the minimum of the total supply of token 1 divided by how much is in the pool, OR, the total supply of token 2 divided by how much is in the pool.

Now you know more about Uniswap than 99% of the people who use it. Most people never bother with this stuff.

If you need to learn about things like impermanent loss (and if you plan to stake in Uniswap's liquidity pools, you really do) you'd do much better to google it and find someone's clear explanation than to try to piece it together from a spreadsheet. It's not intuitive. 

In summary, I'd say this spreadsheet is more useful for someone who already mostly understands Uniswap to grasp some technical points about the economics of it, than as a way for someone who doesn't understand it yet to learn it. The spaghetti pile of calculations under the hood aren't pretty and no better documented than the UI, plus, most of the people who use Uniswap every day don't know them and don't need to.

However, if there's anything that's not conceptually clear from the UI (and can't be easily found by googling), ask away, and I'd be happy to answer. Another great resource, if you really want a programmer's-eye view of Uniswap, is to get on their discord and ask the mods questions, which is how I learned everything I know about all this.

**Comment by beechfuzz** [comment]
Change the extension of the `.xlsx` file to `.zip`, and extract the contents.  Do the following for the extracted `xl\workbook.xml` and all `xl\worksheets\*.xml` files:

- Open the .xml file in notepad.
- Look for a `<sheetProtection ... />` directive 
- Sandwich it between `<!--` and `-->` so that it looks something like this: 
  - `<!--<sheetProtection ... />-->`
- Example:
  - Before:
    - `<sheetProtection algorithmName="SHA-512" hashValue="F1VFLGNUJL3+BcMLwiHZAxEs0DidZlRVuziSHwmUlmrNJvpiOl5SsUH1mXLVijh47nY8yS6LSD2rviycuRFVDA==" saltValue="pAkMrj3A8LA1wMFqODgvDA==" spinCount="100000" sheet="1" objects="1" scenarios="1" selectLockedCells="1"/>`
  - After:
    - `<!--<sheetProtection algorithmName="SHA-512" hashValue="F1VFLGNUJL3+BcMLwiHZAxEs0DidZlRVuziSHwmUlmrNJvpiOl5SsUH1mXLVijh47nY8yS6LSD2rviycuRFVDA==" saltValue="pAkMrj3A8LA1wMFqODgvDA==" spinCount="100000" sheet="1" objects="1" scenarios="1" selectLockedCells="1"/>-->`
- Save the .xml file

Then, re-zip the extracted files into a new `.zip` file and then change the extension of the new `.zip` file to `.xlsx`.  Enjoy your password-unprotected Excel file.

**Comment by ApopheniaPays** [comment]
> Then, re-zip the extracted files into a new `.zip` file and then change the extension of the new `.zip` file to `.xlsx`. Enjoy your password-unprotected Excel file.

Funny I was just coming here to post a new, fully open-source version of the file without any password lock on it at all. I came to this thread to let OP know it was available. And then I saw this incredible dick move from you. 

I've never stolen anything from you, I'm not sure why you think it's ok to come to my personal repo and tell people they should "enjoy" violating my personal terms of use for my personal software that I spent weeks of my personal time writing and then chose to share for anyone to use for absolutely free.

So, since you decided to repay my efforts by being such a dick to me, instead of posting the unlocked open source version, I decided not to share this software at all anymore. I've deleted it from this repo. I hope you're proud of yourself, going out of your way to be a huge dick to me really accomplished something!

Even if you for some reason thought you were entitled to fuller access to something that took me weeks to create and you didn't pay a cent for, you could have cracked the password for yourself (if your instructions actually even worked, which they don't, you wind up with an unopenable file, idiot) and been content. 

But no, that wasn't being enough of a dick for you! You had to come to my repo too and antagonize me to my face, obnoxiously post telling other users they should violate the my terms for my software that I spent my time writing and then let everyone use for free, where you were sure I'd see it. 

Enjoy the new report I just filed against you with Github for using the issues threads to encourage cracking and license violations, enjoy your new being blocked from all my repos, and enjoy everyone else not getting to use this at all anymore because you were had to be such an infantile, entitled little prick towards me. 

Thanks for letting me know not to share my work publicly on Github anymore. I have no desire to do anything generous for infantile little pricks. GFY.

**Comment by ApopheniaPays** [comment]
@beechfuzz:

> Then, re-zip the extracted files into a new `.zip` file and then change the extension of the new `.zip` file to `.xlsx`. Enjoy your password-unprotected Excel file.

Funny I was just coming here to post a new, fully open-source version of the file without any password lock on it at all. I came to this thread to let OP know it was available. And then I saw this incredible dick move from you.

I've never stolen anything from you, I'm not sure why you think it's ok to come to my personal repo to tell people they should "enjoy" violating my terms for my personal software that I spent weeks of my personal time writing and then chose to allow everyone to use for absolutely free.

So, instead of posting the unlocked open source version, since you decided to repay my generosity by going out of your way to intentionally make me angry and act like such a dick right to my face, I decided not to share this liquidity pool simulator at all anymore. I've deleted the file from this repo. I hope you're proud of yourself, going out of your way to be such an entitled dick to me really accomplished something positive!

Even if you for some reason thought you were entitled to fuller access to something that took me weeks to create and you didn't pay a cent for, you could have cracked the password for yourself and been content (if your instructions actually even worked, which they don't, they just create a corrupt file that Excel can't open.)

But no, that wasn't being enough of a dick for you, you weren't satisfied! You had to come to my repo too and antagonize me to my face, obnoxiously post to other users that they should violate the my terms for my software that I spent weeks of my time developing and then let everyone use for free, right here where you were SURE I'd see it. 

Enjoy your report I just filed against you with Github for encouraging password cracking and license violations, enjoy your being blocked from all my repos, and enjoy the entire rest of the world losing access to a project that people were interested in because you just had to go out of your way to be such a prick to the developer, you obnoxious creep. GFY.

Thanks for letting me know not to share my work publicly on Github anymore. I have no desire to do anything generous for infantile little pricks.

**Comment by ApopheniaPays** [comment]
And by the way, @beechfuzz, since you seem to have some sort of infantile oppositional disorder: yes, I'm aware it's possible to recover a deleted github file from a previous commit, and with some determination anyone can still get the file. If you're so insecure that you need to do that to feel good about yourself, there's nothing I can do about it, so go on and kid yourself that spending more your time on stealing my work somehow redeems you. Go on and demean yourself a second time, be even more of a creep, I'm aware there's nothing I can do about it.

**Comment by ApopheniaPays** [comment]
And by the way, @beechfuzz, since you seem to have some sort of infantile oppositional disorder: yes, I'm aware it's possible to recover a deleted github file from a previous commit, and with some determination anyone can still get the file. If you're so insecure that you need to do that to feel good about yourself, there's nothing I can do about it, so go on and kid yourself that spending more your time on stealing my work somehow redeems you. Go on and demean yourself a second time, be even more of a creep, I'm aware there's nothing I can do about it.

**Comment by lowgrind** [comment]
jeeze

**Comment by kupietools** [comment]
Hello, everybody, although interest in this file has died down over time, I just thought I'd mention here that the full liquidity pool simulator excel file has just been re-added to the repo, now unlocked and completely open-source, under a modified GPL3 so it's legally available for everyone except for user @beechfuzz, who is specifically excluded by the license terms.

**Comment by beechfuzz** [comment]
I'm sorry that directions on modifying Excel files without the use of any third-party tools, which are detailed on Microsoft's own forum (https://techcommunity.microsoft.com) hurt your feelings, but your 21-month grudge is more indicative of infantile oppositional disorder than my re-posting of information that is already globally available on Microsoft's own website.

I wish you well in all of your future projects.

**Comment by kupietools** [comment]
@beechfuzz Apparently it wasn't too infantile for you to let go without needing to try to have the last word.
